Makefile

```diff
@@ -1,82 +1,93 @@
-CFLAGS=-g 
+CC=g++
+CFLAGS=-g
+SERVER=server
+EMULATOR=msp430-emu
 
-all: MSP430 SERVER
+.PHONY: all test
 
-MSP430 : main.o utilities.o emu_server.o registers.o memspace.o debugger.o disassembler.o \
-	register_display.o decoder.o flag_handler.o formatI.o formatII.o formatIII.o \
-	usci.o port1.o packet_queue.o bcm.o timer_a.o
+all: ${EMULATOR} ${SERVER}
 
-	g++ ${CFLAGS} -o MSP430 main.o emu_server.o utilities.o registers.o memspace.o debugger.o disassembler.o \
-	register_display.o decoder.o flag_handler.o formatI.o formatII.o formatIII.o usci.o port1.o bcm.o timer_a.o packet_queue.o -lreadline -lwebsockets -lpthread -lrt -lssl -lcrypto;
+# Main emulator program
+
+${EMULATOR} : main.o emu_server.o utilities.o registers.o memspace.o debugger.o disassembler.o \
+	register_display.o decoder.o flag_handler.o formatI.o formatII.o formatIII.o usci.o port1.o bcm.o \
+	timer_a.o packet_queue.o io.o
+	${CC} ${CFLAGS} -o $@ $^ -lreadline -lwebsockets -lpthread -lrt -lssl -lcrypto;
 
 main.o : main.cpp
```

Comment:
Dependencies. `main.cpp` includes `main.h`. In case of change `main.h`, the `main.cpp` file will not be recompiled.
This concerns also other files.

Reply:
The include graph is much more complicated... untangling it is unfeasible with a major refactor. And without untangling this does not change much,,,

Reply:
I think I misunderstood the comment. Now the o files have both c and h files as their dependencies